### PR TITLE
Docs add var len obs to readme and overview

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

This PR adds a section to the ML-Agents overview describing variable length observation and removes some old bullets from the list of feature in the main ML-Agents readme.

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)

[This slack message](https://unity.slack.com/archives/C8FECS6L9/p1613626336229100)
[This JIRA ticket](https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstleyVEVO)

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [ ] Other (please describe)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,17 +26,14 @@ developer communities.
 
 ## Features
 
-- 15+ [example Unity environments](docs/Learning-Environment-Examples.md)
-- Support for multiple environment configurations and training scenarios
```
Comment (mmattar) [inline README.md:30]: Why delete lines 30, 31 and 39?

File: docs/ML-Agents-Overview.md [hunk 2]
```diff
@@ -684,6 +684,23 @@ three network architectures:
 The choice of the architecture depends on the visual complexity of the scene and
 the available computational resources.
 
+### Learning from Variable Length Observations using Attention
+
+Using the ML-Agents Toolkit, it is possible to have agents learn from a
```
Comment (mmattar) [inline docs/ML-Agents-Overview.md:689]: I think this section should start with an example of where you may have different inputs to help a non-RL expert understand when and why they want to use it. (See subsection below for instance.)

File: docs/ML-Agents-Overview.md [hunk 3]
```diff
@@ -684,6 +684,23 @@ three network architectures:
...
+varying number of inputs. To do so, each agent can keep track of a buffer
+of vector observations. At each step, the agent will go through all the
+elements in the buffer and extract information but the elements
+in the buffer can change at every step.
+This can be useful in scenarios in which the agents must keep track of
+a varying number of elements throughout the episode. You can learn more
+about variable length observations and the BufferSensor
+[here](Learning-Environment-Design-Agents.md#variable-length-observations).
+When variable length observations are utilized, the ML-Agents Toolkit
+leverages attention networks to learn from a varying number of entities.
```
Comment (mmattar) [inline docs/ML-Agents-Overview.md:699]: add a reference to attention networks.

File: docs/ML-Agents-Overview.md [hunk 4]
```diff
@@ -684,6 +684,23 @@ three network architectures:
 The choice of the architecture depends on the visual complexity of the scene and
 the available computational resources.
 
+### Learning from Variable Length Observations using Attention
```
Comment (mmattar) [inline docs/ML-Agents-Overview.md:687]: - need to update the Toc above with a link to this section.
- any visuals we can add?

File: README.md (hunk 1 above)
Comment (vincentpierre) [reply README.md:30]: Because I did not like them anymore :) 
I can put them back

File: docs/ML-Agents-Overview.md (hunk 4 above)
Comment (vincentpierre) [reply docs/ML-Agents-Overview.md:687]: I can try to make another comic book

File: docs/ML-Agents-Overview.md (hunk 3 above)
Comment (vincentpierre) [reply docs/ML-Agents-Overview.md:699]: There is a reference already in the section Learning-Environment-Design-Agents.md#variable-length-observations I think we should minimize the number of places with external references and avoid duplicate references.

File: README.md (hunk 1 above)
Comment (mmattar) [reply README.md:30]: ha - let's put them back.

File: docs/ML-Agents-Overview.md (hunk 4 above)
Comment (mmattar) [reply docs/ML-Agents-Overview.md:688]: +1 for comic book.

File: docs/ML-Agents-Overview.md (hunk 3 above)
Comment (mmattar) [reply docs/ML-Agents-Overview.md:705]: Oks oks.

**Review by mmattar — APPROVED** [review]
asked to add back Features... but ow love it.

**Pull request merged**